workflow/scripts/intersection.py

```diff
@@ -15,69 +16,104 @@
 import rasterio
 import pyproj
 from pyproj import Geod
-from snail.core.intersections import get_cell_indices, split_linestring
+from snail.core.intersections import split_linestring
+from snail.core.intersections import get_cell_indices as get_cell_indicies_of_midpoint
 from tqdm import tqdm
 
+from open_gira import fields
 
-def main(network_edges_path, hazard_tifs, output_path):
+
+def associate_raster(df, fname, band_number=1) -> pandas.Series:
     """
-    Split the entries in network_edges_path according to the cells they occupy in the
-    grids of the hazard_dir. Write the results to a .geoparquet file (output_path)
-    and a similarly-named .parquet file.
-
-    Parameters
-    ----------
-    network_edges_path (str): Path to a .geoparquet file with network data
-    attrs (str|List[str]): attribute/s to copy from the original rows when split
-    hazard_dir (List[str]): list of hazard raster files whose values should be combined with the network
-    output_path (str): .geoparquet version of the path to write to. A .parquet version is also written
-
-    Returns
-    -------
-    (void)
+    For each split geometry, lookup the relevant raster value. Cell indicies
+    must have been previously calculated and stored as fields.RASTER_{I,J}.
+
+    N.B. This will store no data values in the returned dataframe.
     """
+    with rasterio.open(fname) as dataset:
+
+        band_data: np.ndarray = dataset.read(band_number)
 
-    hazard_tifs_basenames = [
-        re.sub("\\.tif$", "", os.path.basename(tif)) for tif in hazard_tifs
+        # 2D numpy indexing is j, i (i.e. row, column)
+        return df.apply(
+            lambda row: band_data[row[fields.RASTER_J], row[fields.RASTER_I]],
+            axis="columns"
+        )
```

Comment:
This might not be a particularly time-consuming step, but I wonder if using the columns as index arrays to pick from `band_data` would be faster and neater than the `df.apply`

Reply:
I have no idea if it's faster, but I've just made the change to index with the whole arrays. The kind of change that makes me really happy to have (at least a minimal) test suite to check for regressions.